prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt

```diff
@@ -139,9 +142,28 @@ class ProcessFhirCommands : CliktCommand(
         // if a hl7 parsing failure happens, throw error and show the message
         if (message.toString().lowercase().contains("failed"))
             throw CliktError("HL7 parser failure. $message")
+        if (hasFiveEncodingChars) {
+            val msh = message.get("MSH") as Segment
+            Terser.set(msh, 2, 0, 1, 1, "^~\\&#")
+        }
         return HL7toFhirTranslator.getInstance().translate(message)
     }
 
+    /**
+     * @return true if a message header (either the one at hl7ItemIndex or the first one if hl7ItemIndex is null) in the
+     * given string contains MSH-2 of `^~\&#`, false otherwise
+     */
+    private fun hl7MessageHasFiveEncodingChars(hl7String: String): Boolean {
+        // This regex should match `MSH|^~\&|` or `MSH|^~\&#`
+        val mshStarts = "MSH\\|\\^~\\\\\\&[#|]".toRegex().findAll(hl7String)
+        mshStarts.forEachIndexed { index, matchResult ->
+            if (hl7ItemIndex == null || hl7ItemIndex == index) {
```

Comment:
Should this short circuit if index is null?

Reply:
Tweaked this to make it more readable.